main.tf

```diff
@@ -18,10 +18,24 @@ resource "azurerm_mssql_server" "vaconfigured" {
   version                      = "12.0"
   administrator_login          = "4dm1n157r470r"
   administrator_login_password = "4-v3ry-53cr37-p455w0rd"
-  minimum_tls_version          = "1.1"
+  minimum_tls_version          = "1.2"
 
   transparent_data_encryption_key_vault_key_id = azurerm_key_vault_key.vaconfigured.id
 
+
+  extended_auditing_policy {
```

Comment:
**Tenable.cs remediation**
 
**[LOW]: Enable Role Based Access Control for Azure Key Vault**
 
**Vulnerability**
Description:

WARNING: Role assignments disappear when a Key Vault has been deleted (soft-delete) and recovered. Afterwards it will be required to recreate all role assignments. This is a limitation of the soft-delete feature across all Azure services.

Rationale:

The new RBAC permissions model for Key Vaults enables a much finer grained access control for key vault secrets, keys, certificates, etc., than the vault access policy. This in turn will permit the use of privileged identity management over these roles, thus securing the key vaults with JIT Access management.

Implementation needs to be properly designed from the ground up, as this is a fundamental change to the way key vaults are accessed/managed. Changing permissions to key vaults will result in loss of service as permissions are re-applied. For the least amount of downtime, map your current groups and users to their corresponding permission needs.
 
**Remediation**
**From Azure Portal**

Key Vaults on creation can be set to use `Azure role-based control`.

For already create Key Vaults :

1. From Azure Home open the Portal Menu in the top left corner.
2. Select Key Vaults
3. View the Key Vaults within your subscription.
4. Open every Key Vault you wish to audit.
5. Select Access Policies
6. Set the Permission Model radio button should be set to `Azure role-based control`.
7. In the screen note the warning message, then click save if you have determined the impact.
8. Select `Access Control (IAM)`.
9. Select the `Role Assignments` in the horizontal row.
10. Reapply permissions as needed to groups or users.

source: tenablecs-